libs/wingc/src/type_check.rs

```diff
@@ -1110,7 +1124,7 @@ impl<'a> TypeChecker<'a> {
 		}
 
 		// Verify that all non-optional fields are present and are of the right type
-		for (k, v) in expected_struct.env.iter().map(|(k, v)| {
+		for (k, v) in expected_struct.env.iter(true).map(|(k, v, _)| {
```

Comment:
WDYT of `iter_with_ancestry()` instead of `iter(true)`? Feels slightly more readable to me

Reply:
It's just that the default hsould be to iter with the ancestry. I could create a `iter_without_ancestry` but that feels clunky.

Reply:
Where are we iterating without ancestry?

Reply:
> Where are we iterating without ancestry?

I guess that nowhere... I did this at some point to handle different inheritance related issues. For example you only want to initialize the current resource client's class fields in `init` and you need to pass all other `init` arguments to the client classes `super`. Another case is just to make sure you're jsifying the current resources methods and assuming parent class methods were already jsifyied.
In some places where I just iter over the AST and not the env this isn't an issue. In other places I eventually iterred with ancestry and filtered out the current classes stuff from the ancestors stuff because I acutally neeeded both.
I still think we'll eventually need this so I'm hesitant whether to remove it.